core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -327,16 +328,37 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
             case KOTLIN_COROUTINES:
```

Comment:
Question: Do These changes also solve the Coroutines issue mentioned in #3690?
If so, could you add a test for it?

Reply:
It does seem to work as expected in my local build.
But, would you give me some advice where should I place the test? Maybe in `kotlin` module?

Reply:
You can add tests any place under https://github.com/line/armeria/blob/97ee35c59560d44ab4529ba5d6916f47c572ec03/kotlin/src/test/kotlin/com/linecorp/armeria/server/ :-)

Reply:
Test added!